upstairs/src/downstairs.rs

```diff
@@ -2705,30 +2720,23 @@ impl Downstairs {
         for i in ClientId::iter() {
             match self.clients[i].state() {
                 DsState::LiveRepair => {
-                    self.skip_all_jobs(i);
-                    self.clients[i].abort_repair(up_state, true);
-                }
-                DsState::Faulted => {
-                    // Jobs were already skipped when we hit the IO error that
-                    // marked us as faulted
-                    self.clients[i].abort_repair(up_state, false);
+                    self.fault_client(
+                        i,
+                        up_state,
+                        ClientFaultReason::FailedLiveRepair,
+                    );
```

Comment:
If we no longer need the `Faulted` check here, it means when this DS went to Faulted, it must have triggered the `fault_client()` path.  That would triggered the `skip_all_jobs()` call.

I don't believe that was always true, but it (as far as I can tell) is true now.

Reply:
I think I agree with you – the `DsState::Faulted` check is now in the catchall `_ => { .. }` branch below.  You don't want anything to change here, right?